src/test/lib/operators/table_scan_sorted_segment_search_test.cpp

```diff
@@ -78,57 +92,99 @@ INSTANTIATE_TEST_SUITE_P(
+            TestData{"BetweenLowerExclusive", PredicateCondition::BetweenLowerExclusive, 0, 3, {1, 1, 2, 2, 3, 3}},
+            TestData{"BetweenLowerExclusiveAboveRange", PredicateCondition::BetweenLowerExclusive, 4, 10, {}},
+            TestData{"BetweenLowerExclusiveBelowRange", PredicateCondition::BetweenLowerExclusive, -5, -1, {}},
+
+            TestData{"BetweenUpperExclusiveWiderRange", PredicateCondition::BetweenUpperExclusive, -1, 6, {0, 0, 1, 1, 2, 2, 3, 3, 4, 4}},  // NOLINT
+            TestData{"BetweenUpperExclusiveExactRange", PredicateCondition::BetweenUpperExclusive, 0, 5, {0, 0, 1, 1, 2, 2, 3, 3, 4, 4}},  // NOLINT
+            TestData{"BetweenUpperExclusiveRangeMinimum", PredicateCondition::BetweenUpperExclusive, 0, 4, {0, 0, 1, 1, 2, 2, 3, 3}},  // NOLINT
+            TestData{"BetweenUpperExclusiveRangeMaximum", PredicateCondition::BetweenUpperExclusive, 2, 5, {2, 2, 3, 3, 4, 4}},  // NOLINT
+            TestData{"BetweenUpperExclusive", PredicateCondition::BetweenUpperExclusive, 1, 4, {1, 1, 2, 2, 3, 3}},
+            TestData{"BetweenUpperExclusiveAboveRange", PredicateCondition::BetweenUpperExclusive, 5, 10, {}},
+            TestData{"BetweenUpperExclusiveBelowRange", PredicateCondition::BetweenUpperExclusive, -5, 0, {}}),
 
         ::testing::Values(SortMode::Ascending, SortMode::Descending),
-        ::testing::Bool()),  // nullable
+        ::testing::Values(NullValueUsage::WithoutNulls, NullValueUsage::WithNulls, NullValueUsage::OnlyNulls)),
     table_scan_sorted_segment_search_test_formatter);
 // clang-format on
 
 TEST_P(OperatorsTableScanSortedSegmentSearchTest, ScanSortedSegment) {
   const auto iterable = create_iterable_from_segment(*_segment);
   iterable.with_iterators([&](auto input_begin, auto input_end) {
     auto sorted_segment_search =
-        SortedSegmentSearch(input_begin, input_end, _sorted_by, _nullable, _predicate_condition, _search_value);
-    sorted_segment_search.scan_sorted_segment([&](auto output_begin, auto output_end) {
-      ASSERT_EQ(std::distance(output_begin, output_end), _expected.size());
-
-      size_t index = 0;
-      for (; output_begin < output_end; ++output_begin, ++index) {
-        ASSERT_FALSE(output_begin->is_null()) << "row " << index << " is null";
-        ASSERT_EQ(output_begin->value(), _expected[index]) << "row " << index << " is invalid";
-      }
-    });
+        _second_search_value
+            ? SortedSegmentSearch(input_begin, input_end, _sorted_by, _nullable, _predicate_condition, _search_value,
+                                  *_second_search_value)
+            : SortedSegmentSearch(input_begin, input_end, _sorted_by, _nullable, _predicate_condition, _search_value);
+    auto matches = RowIDPosList{};
+    const auto initial_chunk_id = ChunkID{0};
+    sorted_segment_search.scan_sorted_segment(initial_chunk_id, matches, nullptr);
```

Comment:
Do we test the early out for "all NULLs"?

Reply:
Good catch, gonna check that `no_rows_matching` is set.

Reply:
I added checks that `all_rows_matching` is set when `matches.size() == _expected.size()` and `no_rows_matching` is set when `matches.empty()`

Reply:
Nice.

And is there a test case that explicitely tests for all values being NULL? I haven't found this case when skipping over this test file here.

Reply:
I changed the testing parameter that switches the segment's nullability from a bool to an enum with three values. One of these enum values implies that all values should be NULL and non-null values are not appended to the segment. When this flag is set, we do not check whether the scan result is equal to the expected test data but check if the scan returns an empty PosList.